sharepoint-adfs/azuredeploy.json

```diff
@@ -621,14 +715,29 @@
         ]
       }
     },
+    {
+      "name": "[variables('vmSP').vmDataDiskName]",
```

Comment:
Any reason we can't use implicit disks?

Reply:
Can you please explain what is an implicit disk?

Reply:
An implicit managed disk is created automatically, if an explicit one is not specified - see: https://github.com/Azure/azure-quickstart-templates/blob/master/201-vm-new-or-existing-conditions/azuredeploy.json#L269-L272

Reply:
Ok, I think I get it, but not with the full picture.
If I understand correctly, I use implicit disks with OS disks but not with data disks (that reference an explicit "Microsoft.Compute/disks" resource).
From what I remember, the reason I used explicit data disks was that this was the only way I found to specify a meaningful name, which improves readability when you look at the resources in the resource group.

Reply:
You should be able to name an implicit disk...  just add the name property

Reply:
I can use implicit disk for OS disks, but I can't get it to work with dataDisks:
- In Microsoft.Compute/disks / dataDisks / managedDisk: If I omit the parameter Id, deployment fails
- In Microsoft.Compute/disks / dataDisks : If I omit the node managedDisk, deployment fails
I really doubt it's possible to add an implicit disk in dataDisks, I couldn't find any example of this

Reply:
Just remove the disk resources and remove the ID on the managedDisk property in the VM - you can name them if you need to, see: https://docs.microsoft.com/en-us/azure/templates/microsoft.compute/virtualmachines

Reply:
I removed the disk resource entirely, and removed the key "Id" in "managedDisk":
```json
"dataDisks": [
	{
	  "lun": 0,
	  "name": "[variables('vmSP').vmDataDiskName]",
	  "createOption": "Attach",
	  "caching": "ReadWrite",
	  "managedDisk": {
	  }
	}
]
```
It fails with this error:
```
New-AzureRmResourceGroupDeployment : 13:58:28 - Resource Microsoft.Compute/virtualMachines 'SP' failed with message '{
  "error": {
    "code": "InvalidParameter",
    "target": "dataDisk.managedDisk.id",
    "message": "Required parameter 'dataDisk.managedDisk.id' is missing (null)."
  }
}'
```

I get the same error if I also remove the key "name".
I tried quite a few times but I'm not able to create a managed data disk without creating an explicit disk resource explicitly (regardless of giving it a meaningful name).
I have the feeling that this is not possible, but I understand you are certain this can be done.
Can you see what I'm doing wrong?

Reply:
Use

"createOption": "empty"

Reply:
Yes it works!
Thank you for your help and patience on this, I feel so bad I missed that for so long...